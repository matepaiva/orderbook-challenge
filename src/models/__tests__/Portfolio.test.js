const { Document } = require('camo');
const { Portfolio } = require('../index');

jest.mock('../modifiers', () => ({
    withTimestamps: (val) => val,
}));

jest.mock('camo', () => ({
    Document: class MockDocument { },
}));


describe('Portfolio', () => {
    describe('.constructor()', () => {
       it('should define fields', () => {
           const portfolio = new Portfolio();

           expect(portfolio).toMatchSnapshot();
           expect(portfolio).toBeInstanceOf(Document);
       });
    });

    describe('.collectionName()', () => {
        it('should return the collection name', () => {
            expect(Portfolio.collectionName()).toBe('portfolios');
        });
    });
});
