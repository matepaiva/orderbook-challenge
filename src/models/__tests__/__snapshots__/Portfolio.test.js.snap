// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio .constructor() should define fields 1`] = `
Portfolio {
  "fundAmount": Object {
    "min": 1000,
    "required": true,
    "type": [Function],
  },
}
`;
